service/controller/resource/alert/resource.go

```diff
@@ -40,6 +39,7 @@ func New(config Config) (*generic.Resource, error) {
 		ClientFunc:     clientFunc,
 		Logger:         config.Logger,
 		Name:           Name,
+		GetObjectMeta:  rules.GetObjectMeta,
 		ToCR:           toPrometheusRule,
```

Comment:
```suggestion
		ToCR:           rules.ExampleRule,
```

Reply:
Good point, done.

Reply:
Hmm, actually this causes:

```
# github.com/giantswarm/prometheus-meta-operator/service/controller/resource/alert
service/controller/resource/alert/resource.go:43:3: cannot use rules.ExampleRule (type func(interface {}) (*"github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1".PrometheusRule, error)) as type func(interface {}) ("k8s.io/apimachinery/pkg/apis/meta/v1".Object, error) in field value
```

any ideas on how to handle this better?

Reply:
change the signature to `func(interface{}) (metav1.Object, error)`